src/ploneintranet/workspace/adapters.py

```diff
@@ -271,34 +271,64 @@ def metromap_sequence(self):
+        except api.exc.UserNotFoundError:
+            raise api.exc.UserNotFoundError(
+                "Unknown user. Do not use Zope rescue user.")
         current_state = wft.getInfoFor(self.context, "review_state")
-        next_available_transition = [
-            i.get('next_transition') for i in metromap_list
-            if i['state'] == current_state and 'next_transition' in i
-        ]
         finished = True
         sequence = OrderedDict()
-        for index, mmap in enumerate(metromap_list):
-            state = mmap['state']
+        tasks = self.context.tasks()
+        for index, wfstep in enumerate(metromap_list):
+            state = wfstep['state']
             if state == current_state:
-                finished = False
-            next_transition = mmap.get('next_transition')
-            reopen = False
-            if index + 1 < len(metromap_list):
+                is_current = True
+                finished = False  # keep this for the rest of the loop
+                open_tasks = [x for x in tasks[state] if not x['checked']]
+            else:
+                is_current = False
+                open_tasks = []  # we don't care so performance optimize
+
+            # last workflow step: consider done if no open tasks left
+            if (state == current_state
+               and index > len(metromap_list)
+               and not open_tasks):
+                finished = True
+
+            # only current state can be closed
+            if (state == current_state and can_manage and not open_tasks):
+                next_transition = wfstep.get('next_transition', None)
+            else:
+                next_transition = None
+            if next_transition:
+                transition_title = _(
+                    cwf.transitions.get(next_transition).title)
+            else:
+                transition_title = ''
+
+            # reopen only the before-current step, only for admins
+            reopen_transition = None
+            try:
                 next_state = metromap_list[index + 1]["state"]
-                reopen = next_state == current_state
+                # if this step precedes the current state, it can be reopened
+                if next_state == current_state and can_manage:
+                    reopen_transition = wfstep.get('reopen_transition', None)
+            except IndexError:
+                # last step, no next
+                pass
+
             sequence[state] = {
                 'title': _(cwf.states.get(state).title),
                 'transition_id': next_transition,
-                'transition_title': '',
-                'reopen_transition': reopen and mmap['reopen_transition'],
-                'enabled': next_transition in next_available_transition,
+                'transition_title': transition_title,
+                'reopen_transition': reopen_transition,
+                'is_current': is_current,
                 'finished': finished,
```

Comment:
This replaces 'enabled' with 'is_current'. @cillianderoiste What I've done in Plone Intranet, you may need to do in ikath:
`grep -r enabled . | cut -f1 -d: | grep -v Binary | uniq | xargs sed -i -e 's/enabled/is_current/g'`